lib/logstash/outputs/rabbitmq.rb

```diff
@@ -142,18 +141,29 @@ def connect
     channel = connection.create_channel
     @logger.info("Connected to RabbitMQ at #{settings[:host]}")
 
+    exchange = declare_exchange!(channel)
+
+    HareInfo.new(connection, channel, exchange)
+  end
+
+  private
+  def declare_exchange!(channel)
     @logger.debug("Declaring an exchange", :name => @exchange,
                   :type => @exchange_type, :durable => @durable)
-
     exchange = channel.exchange(@exchange, :type => @exchange_type.to_sym, :durable => @durable)
     @logger.debug("Exchange declared")
-
-    HareInfo.new(connection, channel, exchange)
+    exchange
+  rescue StandardError => e
+    @logger.error("Could not declare exchange!",
+                  :exchange => @exchange, :type => @exchange_type,
+                  :durable => @durable, :error_class => e.class.name,
+                  :error_message => e.message, :backtrace => e.backtrace)
+    raise e
   end
 
   private
   def connect!
-    @hare_info = connect() unless connection_open?
+    @hare_info = connect() unless @hare_info
```

Comment:
Since we are removing `connect` from the retry logic, I don't see the need to have two similar methods?  `connect` and `connect!` because the last one doesn't do much.